analyzers/tests/SonarAnalyzer.UnitTest/TestCases/VariableShadowsField.CSharp12.cs

```diff
@@ -0,0 +1,14 @@
+﻿class MyClass(int first, int second, int third)
+{
+    private int first; // Compliant
+
+    void MyMethod()
+    {
+        int second = 1; // Compliant
```

Comment:
I would add some non-compliant scenarios, such as a local variable `int first = 1`, hiding the primary constructor parameter.

Reply:
I don't understand this request, theoretically the variable `second` is already hiding the primary constructor parameter. To create a non-compliant example we should create a `first` variable inside the method, but this is already covered by the base examples

Reply:
Shadowing here is a bit subtle: in the previous version of the code, `second = 1` was not shadowing the primary constructor parameter, which would be turned into an unspeakable field at compilation. It was shadowing the field with the same name.
Compare [this example](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKAKwAIBhAClgBdiBnAUwGMB7GBASgIG8Di/jqdJqwTEAvEJZsA3AV790AFmIBZAJ4r6VABbME5Tvn7EeR4/wZSEyZLLPni8h6YcPBlkeNUQdAOgCCwLTkiux2ruYebDbhrgC+TsQJ+HFAA), where the shadowed field is `second`, with [this one](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKAKwAIBhAClgBdiBnAUwGMB7GBASgIG8Di/iA9AOp0mrBMQC8olmwDcBXv3QAWYgFkAnuvpUAFswTlO+fsR6mz/BrITJkCy1eJLnF585E3xUjRH0AdACCwLTkKuyOHlbebPZRHgC+rsTJ+IlAA), where the shadowed is `<second>P`.
Because the creation of an unspeakable field, when the parameter of a primary constructor is used in a method, is not defined in the spec, and it is rather an implementation detail subject to change.

Moreover, when a primary constructor parameter is not "promoted" to a non-unspeakable field (i.e. it is not explicitly assigned to a user-declared field with the same name), shadowing from parent class works a bit in mysterious ways: see [here](https://github.com/dotnet/csharplang/discussions/7109#discussioncomment-5671145) for an example, that shows this [interesting sharplab example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAUEEsC2UECeAwgPYB2yMCcAxjCQsgDQwgBuE2YBTAJiAGoAPgAEAzAAIRAJgkAhKDQhEANlGTIJAWABQAb10SjUyVjIwJADwDcugL67d4iUhoNe8xRABKENwl5dAx1jEwkzCxt7Rx1nGQkiAAoIqwBKCRBPJVV1TSDDY2cUgFlE9IBeAD4ra20dBx0nSVd3CW9k8zTMhSVff0D9AqMiztKK6ps6hqA=).

The new version of your code still tests only for user-defined field shadowing. You can test for direct primary constructor parameter shadowing by simply removing the `int second` field;

Reply:
As discussed offline, the test is correct and already captures the scenario that I though was not captured.
Resolving.